src/sdk/clients/createNexusSessionClient.ts

```diff
@@ -0,0 +1,193 @@
+  createWalletClient,
+  encodePacked
+} from "viem"
+import type {
+  PaymasterActions,
+  SmartAccount,
+  UserOperationRequest
+} from "viem/account-abstraction"
+import contracts from "../__contracts"
+import type { Call } from "../account/utils/Types"
+
+import { generatePrivateKey, privateKeyToAccount } from "viem/accounts"
+import type { UnknownSigner } from "../account/utils/toSigner"
+import { toSmartSessionValidatorModule } from "../modules/validators/smartSessionValidator/tosmartSessionValidatorModule"
+import type { ToValidationModuleReturnType } from "../modules/validators/toValidationModule"
+import { type NexusClient, createNexusClient } from "./createNexusClient"
+
+// Review: Marked for removal.
+/**
+ * Parameters for sending a transaction
+ */
+export type SendTransactionParameters = {
+  calls: Call | Call[]
+}
+// UseSessionParameters
+
+/**
+ * Configuration for creating a Session Nexus Client
+ */
+// Review: Could create our own type with only things needed for session client.
+export type NexusSessionClientConfig<
+  transport extends Transport = Transport,
+  chain extends Chain | undefined = Chain | undefined,
+  account extends SmartAccount | undefined = SmartAccount | undefined,
+  client extends Client | undefined = Client | undefined,
+  rpcSchema extends RpcSchema | undefined = undefined
+> = Prettify<
+  Pick<
+    ClientConfig<transport, chain, account, rpcSchema>,
+    | "account"
+    | "cacheTime"
+    | "chain"
+    | "key"
+    | "name"
+    | "pollingInterval"
+    | "rpcSchema"
+  > & {
+    /** RPC URL. */
+    transport: transport
+    /** Bundler URL. */
+    bundlerTransport: transport
+    /** Client that points to an Execution RPC URL. */
+    client: client | Client | undefined
+    /** Paymaster configuration. */
+    paymaster?:
+      | true
+      | {
+          /** Retrieves paymaster-related User Operation properties to be used for sending the User Operation. */
+          getPaymasterData?: PaymasterActions["getPaymasterData"] | undefined
+          /** Retrieves paymaster-related User Operation properties to be used for gas estimation. */
+          getPaymasterStubData?:
+            | PaymasterActions["getPaymasterStubData"]
+            | undefined
+        }
+      | undefined
+    /** Paymaster context to pass to `getPaymasterData` and `getPaymasterStubData` calls. */
+    paymasterContext?: unknown
+    /** User Operation configuration. */
+    userOperation?:
+      | {
+          /** Prepares fee properties for the User Operation request. */
+          estimateFeesPerGas?:
+            | ((parameters: {
+                account: account | SmartAccount
+                bundlerClient: Client
+                userOperation: UserOperationRequest
+              }) => Promise<EstimateFeesPerGasReturnType<"eip1559">>)
+            | undefined
+        }
+      | undefined
+    /** Owner of the account. */
+    // Note: Made the signer optional. This can be a session signer?
```

Comment:
review

Reply:
Yes, we can pass here a session signer but I can see that in "createNexusSessionClient we are already building a signer "incompatibleSigner", isn't that the same ?